src/main/java/apoc/metrics/Metrics.java

```diff
@@ -0,0 +1,107 @@
+package apoc.metrics;
+
+import apoc.ApocConfiguration;
+import apoc.load.LoadCsv;
+import apoc.util.FileUtils;
+import org.neo4j.procedure.Description;
+import org.neo4j.procedure.Mode;
+import org.neo4j.procedure.Name;
+import org.neo4j.procedure.Procedure;
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public class Metrics {
+    /**
+     * DAO for a single line in a metrics/*.csv file.
+     * Value is abstracted to double; in reality some are int, some are float, some are double.
+     */
+    public static class Neo4jMetricEntry {
+        public final long t;
+        public final double value;
+        public final String metric;
+
+        public Neo4jMetricEntry(long t, double value, String metric) {
+            this.t = t;
+            this.value = value;
+            this.metric = metric;
+        }
+    }
+
+    public static class Neo4jMetric {
+        public final String name;
+        public final long lastUpdated;
+
+        public Neo4jMetric(String name, long lastUpdated) {
+            this.name = name;
+            this.lastUpdated = lastUpdated;
+        }
+    }
+
+    @Procedure
+    @Description("apoc.metrics.list() - get a list of available metrics")
+    public Stream<Neo4jMetric> list() {
+        Object metricsSetting = ApocConfiguration.get("dbms.directories.metrics", "");
+
+        File metricsDir = (
+                (metricsSetting == null) ?
+                        new File(ApocConfiguration.get("unsupported.dbms.directories.neo4j_home", ""), "metrics") :
+                        new File(metricsSetting.toString()));
+
+        if (!metricsDir.exists() || !metricsDir.isDirectory() || !metricsDir.canRead()) {
+            return Stream.empty();
+        }
```

Comment:
Perhaps you can extract line 46 to 55 into a function that returns a File or null and return Stream.empty() on null
And then reuse the same function in `apoc.metrics.get`